### PR TITLE
Third attempt at migrating to rehype

### PR DESCRIPTION
**Pull request by iAdramelk**
Because of the revert, there are a lot of files in the PR. To make the review faster, I split every new fix into separate commits and lined them below.

Changes:

- #247 is partially fixed – no need to reload the server anymore, but you still need to reload the pages. Making real hot reload work with `getServerSideProps` may be impossible or pretty hard. Commit: https://github.com/gravitational/docs/pull/253/commits/9e8b8355b789be14eff078c01f2ebbb52487bee8
- #248 - fixes bugs with missing attributes and adds tests for the plugin. Commit: https://github.com/gravitational/docs/pull/253/commits/8124d2b18ec0c595cc61cf3e624d1d066b774f3e
- #161 - fixes image aspect ratio by updating to Next@13. Next@13 reworked the way images are inserted into the page by removing span wrappers and absolute positioning. It fixed a lot of problems with image proportions. To make it even more stable, I replaced `probe-image-size` library with the `image-size` library used by Next.js itself. https://github.com/gravitational/docs/pull/253/commits/9264aacff9a1ca8205c80c557c28f5d225c55aa7
- #243 - Make current version URLs with `/ver/x.x`work via redirects. Commit: https://github.com/gravitational/docs/pull/253/commits/64f39d8e3f31452445ade6cef3ff4ba7d5c51da2
- Fix a minor bug with `Var` component hydration problem. Commit: https://github.com/gravitational/docs/pull/253/commits/22ab61e3ba318f6708efef54433421f4a1a7c933
- Updated webpack to fix vulnerability warning. Commit: https://github.com/gravitational/docs/pull/253/commits/3bb98fc6f86b1b79cc2861882bcd7b25029df2e8
- Return older-versions page. Commit: https://github.com/gravitational/docs/pull/253/commits/6d6e6d5f62102a78e7020048782ea8c4012fc3f0

#246 is not a part of this PR because it is not related to rehype migration. I will work on it separately.

### CODE REVIEW

File: server/paths.ts
Comment (iAdramelk) [reply server/paths.ts:133]: The original `slug` looks like this `/var/13.x/page-name/`, so we need to filter the first and the last blank segments. We could do it in the slug generation function, but then we will need to add them back to the source map and other places.

File: server/paths.ts
Comment (iAdramelk) [reply server/paths.ts:159]: Yes, right now, we push twice, but I am thinking about changing it to rewrites in config instead of actual paths.

File: server/paths.ts
Comment (iAdramelk) [reply server/paths.ts:186]: No, it sets values twice. E. g. `/docs/` and `/docs/ver/12.x/`. But I will also try  the redirect idea so w may not need it.

**Comment by iAdramelk** [comment]
@avatus  Returned `older-versions` page and replaced the current version solution with redirects. Please check

**Comment by iAdramelk** [comment]
@avatus ok, we have a problem, lol. I remember why I removed `older-versions` – we already have it in the docs, so we now have a naming conflict. Will add an exception for now, but we should remove `older-versions.mdx` from docs.

File: server/paths.ts
Comment (avatus) [inline server/paths.ts:128]: We can remove this check once https://github.com/gravitational/teleport/pull/23192 gets pulled in. But good to know we have a killswitch for anything similar in the future. thank you!

**Review by avatus — APPROVED** [review]
Looks good to me and has been working great locally. Would wait for @alexfornuto to go over it as his comb is much more fine than mine

**Review by alexfornuto — CHANGES_REQUESTED** [review]
The Good
- Local refresh - working manually, though it's a bummer that live refresh won't work.
- Video auto playing - confirmed. 
- Images - reviewed several png and svg files, they all appear to render correctly.

The bad
- Versioned paths - Using a versioned link for the current version successfully redirects, but using the version switcher on the page errors out:

  ```
  error - ./content/11.x/docs/pages/deploy-a-cluster/introduction.mdx
  Module parse failed: Assigning to rvalue (1:2)
  You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
  > ---
  | title: "Running a Production Teleport Cluster"
  | description: "Guides to running Teleport in production."
  ```

  This error can't be escaped by navigating to another page, the dev server needs to be restarted.

- Vars - now that #254 is merged, can you rebase this branch against master to confirm that it all works when put together?

**Comment by iAdramelk** [comment]
> error - ./content/11.x/docs/pages/deploy-a-cluster/introduction.mdx
> Module parse failed: Assigning to rvalue (1:2)
> You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> > ---
> | title: "Running a Production Teleport Cluster"
> | description: "Guides to running Teleport in production."

@alexfornuto Can please you check if you have `ver` folder with the mdx files inside the `pages` folder? If yes, can you remove them and try again? They are automatically placed in the folder by the previous build system and can break the new one.

**Comment by iAdramelk** [comment]
BTW, rebased from the main and tested. It works.

**Review by alexfornuto — APPROVED** [review]
> Can please you check if you have `ver` folder with the mdx files inside the `pages` folder? If yes, can you remove them and try again?

👍 

Also tested and confirmed that vars are working. I think this is good, but would like to wait for @ptgott's review as well.

**Comment by ptgott** [comment]
The only issue I've run into: when running `yarn dev`, editing a file, and writing the edits, the page doesn't refresh. I can still show the changes by refreshing manually. @alexfornuto Do you get the same issue and is that okay for you?

Otherwise, I've given the docs a quick browse with the dev server running and everything seems okay!

**Review by avatus — APPROVED** [review]

**Comment by iAdramelk** [comment]
Ready to merge if @alexfornuto is ok with it.

**Comment by alexfornuto** [comment]
@iAdramelk I am begrudgingly OK with it.

**Pull request merged**
